### PR TITLE
feat(ST-47): create and migrate classroom entity

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added classroom entity file
- Added many to one relation with teacher 
- Added classrooms module and repository
- Migrated the Entity

## Associated Ticket Link(s)

- https://trello.com/c/SWGbB6Nz

## Related Pull Request(s)

- N/A

### CODE REVIEW

File: src/common/entities/classrooms.entity.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+})
+export class Classroom extends CustomBaseEntity {
+  [EntityRepositoryType]?: ClassroomRepository;
+
+  constructor(title: string, subject: string, clasTime: Date, days: Array<string>) {
+    super();
+    this.title = title;
+    this.subject = subject;
+    this.classTime = clasTime;
+    this.days = days;
```
Comment (ishmam-iqbal-sazim) [inline src/common/entities/classrooms.entity.ts:27]: Don't need these. Use createOne method to create records

File: src/db/migrations/Migration20240829144904_classroom_field_name_update.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Migration } from '@mikro-orm/migrations';
```
Comment (ishmam-iqbal-sazim) [inline src/db/migrations/Migration20240829144904_classroom_field_name_update.ts:1]: You don't need this migration. Just generate a new one with both changes

File: src/common/entities/classrooms.entity.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+} from "@mikro-orm/core";
+
+import { ClassroomRepository } from "@/classrooms/classrooms.repository";
+
+import { CustomBaseEntity } from "./custom-base.entity";
+import { Teacher } from "./teachers.entity";
+
+@Entity({
+  tableName: "classrooms",
+  repository: () => ClassroomRepository,
```
Comment (ishmam-iqbal-sazim) [inline src/common/entities/classrooms.entity.ts:17]:
```suggestion
  repository: () => ClassroomsRepository,
```

Should be plural

File: src/classrooms/classrooms.repository.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { Injectable } from "@nestjs/common";
+
+import { EntityRepository } from "@mikro-orm/postgresql";
+
+import { Classroom } from "@/common/entities/classrooms.entity";
+
+@Injectable()
+export class ClassroomRepository extends EntityRepository<Classroom> {}
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.repository.ts:8]:
```suggestion
export class ClassroomsRepository extends EntityRepository<Classroom> {}
```

**Review by ragib-hossain-sazim — APPROVED** [review]

**Review by ishmam-iqbal-sazim — APPROVED** [review]

**Pull request merged**